primefaces/src/main/java/org/primefaces/renderkit/SelectRenderer.java

```diff
@@ -43,6 +43,9 @@ public abstract class SelectRenderer extends InputRenderer {
 
     protected boolean isHideNoSelection(UIComponent component) {
         Object attribute = component.getAttributes().get("hideNoSelectionOption");
+        if (attribute instanceof String) {
+            attribute = Boolean.parseBoolean(Objects.toString(attribute));
```

Comment:
Just curious, why didn't you do it like this?
```suggestion
            attribute = Boolean.parseBoolean((String) attribute);
```